### DISCUSSION TITLE
Mode strict

### DISCUSSION THREAD

**Discussion by mquandalle** [publicodes/publicodes#401] [Ideas]
Par défaut un certains nombres d'erreurs sortent des avertissements non bloquants, notamment : 

- règle parente non définie #278
- incohérence d'unités dans une opération #53
- `setSituation` avec un nom de règle inconnue #257

Par ailleur certains autres ne lèvent pas de warning mais sont des comportements que l'on pourrait vouloir rendre plus strict : 
- #202 

Avec un flag `strict: boolean` on peut passer ses problèmes en erreurs bloquantes.

À voir si comme typescript on ajoute des flags individuels `strict-units`, `strict-parent-rule`, `strict-set-situation`, etc.

**Comment by johangirod** [comment]
Les flags individuels peuvent être une bonne idée : sur mon-entreprise, on souhaiterai par exemple activer tous les flags strict sauf celui sur l'utilisation de règles `experimental` dans `setSituation`. On souhaiterai même le désactiver dans notre cas.

**Comment by mquandalle** [comment]
Pas 100% familier des règles `experimental`, mais peut-être que dans ce cas là c'est même au niveau de `engine.evaluate()` que vous voulez une option `accepteExperiementalRule` ou quelque chose comme ça ?

**Comment by johangirod** [comment]
Mmmh pour moi c'est le même cas d'usage que `setSituation`, on ne veut pas polluer la signature de la fonction (d'ailleurs un `setSituation` lance également un warning en cas de règle expérimentale dans l'objet en paramètre). 

`experimental` permet aux créateurs de modèle exposés à des réutilisateur de taguer certaines règles comme étant en dehors du champ de la semver. Cette fonctionnalité n'est pas encore documentée.
